File: be-smart-job/src/main/java/be_smart_job/service/job/interfaces/CategoryService.java
package be_smart_job.service.job.interfaces;

import be_smart_job.dto.req.job.CategoryRequest;
import be_smart_job.dto.res.job.CategoryResponse;

import java.util.List;

public interface CategoryService {
    List<CategoryResponse> getAllCategories();
    CategoryResponse getCategoryById(String id);
    CategoryResponse createCategory(CategoryRequest request);
    CategoryResponse updateCategory(String id, CategoryRequest request);
    void deleteCategory(String id);
}

File: fe-smart-job/src/services/categoryService.js
import axiosClient from '../config/axiosClient';

export const categoryService = {
  getAllCategories: () => axiosClient.get('/categories'),
  
  getCategoryById: (id) => axiosClient.get(`/categories/${id}`),
  
  createCategory: (data) => axiosClient.post('/categories', data),
  
  updateCategory: (id, data) => axiosClient.put(`/categories/${id}`, data),
  
  deleteCategory: (id) => axiosClient.delete(`/categories/${id}`),
};